### ISSUE TITLE
Roadmap Feature Request: LaTeX math support

### ISSUE THREAD

**Issue by marc-medley** [calebkleveter/SwiftMark#5]
Support dollars (`$InlineMath$`, `$$DisplayMath$$`) syntax as Markdown delimiters for LaTeX math expressions. Possibly also add single backslash `\(InlineMath\)`, `\[DisplayMath\]` and double backslash `\\(InlineMath\\)`, `\\[DisplayMath\\]` syntax.  Provide an enable|disable for each implemented syntax.

Such LaTeX math sections would be passed through to html for rendering by a library such as [MathJax](https://www.mathjax.org/) (more comprehensive) or [KaTeX](https://katex.org/) (faster).

Various Markdown editing environments support LaTeX math. (e.g. [Atom](https://atom.io/packages/markdown-preview-enhanced), [MacDown](https://macdown.uranusjr.com/features/), [MWeb](https://www.mweb.im/markdown.html#toc_27)). It would be nice if markdown containing LaTeX math could be processed by SwiftMark. :-)

Aside, but related FYI, Apple has added LaTeX mathematical equation support to Pages, Numbers, Keynote, and iBooks Author. Apple: [HT207569](https://support.apple.com/en-ca/HT207569), [HT202501](https://support.apple.com/en-us/HT202501)

In my use case, dollars are most frequently used in Markdown with the double backslash as a distant second. I don't use single backslash notation except for some testing & evaluation. One such test set is here: [004.55_Markdown-CLI-Evaluation](https://github.com/marc-medley/004.55_Markdown-CLI-Evaluation).

There is also a community syntax survey of [Math in MarkDown on GitHub](https://github.com/cben/mathdown/wiki/math-in-markdown). The [appendix summary table](https://github.com/cben/mathdown/wiki/math-in-markdown#appendix-summary-table) shows the dollars to be the a widely used Markdown math delimiter syntax.
